### PR TITLE
feat(remix-dev): support a TypeScript file for the `remix.init` index script

### PR DESCRIPTION
**Pull request by dvargas92495**
Closes: #
  - Sorry, I just decided to make the PR instead of making a discussion first. I don't mind if the PR gets declined and find PRs a better context for relevant discussion.

- [ ] Docs
    - Happy to add Docs if the maintainers agree with this change
- [x] Tests

The goal for this PR is to support Typescript for the Remix Init Stack script. I plan to have a pretty hefty Init script for my custom stack, and want to avoid writing that in JavaScript.

### CODE REVIEW

File: packages/remix-dev/__tests__/create-test.ts [hunk 1]
```diff
@@ -391,6 +391,22 @@ describe("the create command", () => {
...
   });
 
+  it("runs remix.init script when using index.ts", async () => {
+    let projectDir = await getProjectDir("remix-init-auto");
+    jest.setTimeout(100000);
+    await run([
+      "create",
+      projectDir,
+      "--template",
+      path.join(__dirname, "fixtures", "stack-init-ts.tar.gz"),
```
Comment (dvargas92495) [inline packages/remix-dev/__tests__/create-test.ts:401]: I just copied the stack under `/fixtures/stack`, converting the index file to `index.ts` and console logging the message below in the default method exported

**Comment by machour** [comment]
Hi @dvargas92495 and thank you for your submission!

Can you rebase your PR against the `dev` branch? Code changes need to target that branch.

**Comment by dvargas92495** [comment]
Just rebased it! Sorry for not reading directions 🙈

File: packages/remix-dev/cli/create.ts [hunk 2]
```diff
@@ -188,7 +188,8 @@ async function extractLocalTarball(
     throw Error(
       "🚨 There was a problem extracting the file from the provided template.\n\n" +
         `  Template filepath: \`${filePath}\`\n` +
-        `  Destination directory: \`${projectDir}\``
+        `  Destination directory: \`${projectDir}\`\n` +
+        `  ${err}`
```
Comment (dvargas92495) [inline packages/remix-dev/cli/create.ts:192]: This test is passing locally and failing on this error. Happy to remove this after I find out why it's failing on GH actions, but I think the error itself should be somewhere in this error message

**Comment by dvargas92495** [comment]
hey @machour, could we run tests on this PR?

**Comment by dvargas92495** [comment]
Okay this time I tar's the file with the script and ensure permissions were forgiving before tarring. Can we rerun the tests @machour?

**Comment by dvargas92495** [comment]
any insight on why I'm still getting eaccess permission errors? I'm just copying the existing `/stack` fixture and then running the local `tar.js` script. Could it be related to me running windows locally?

**Comment by dvargas92495** [comment]
@machour @MichaelDeBoey any thoughts on my question above?

**Comment by MichaelDeBoey** [comment]
@dvargas92495 Could you please rebase on latest `dev` & resolve conflicts?

**Comment by dvargas92495** [comment]
I was just able to get the tests passing on my repo: https://github.com/dvargas92495/remix/pull/1, so no I no longer need help getting unblocked.

Going to rebase now and poke for review

**Comment by dvargas92495** [comment]
Ok rebased! @MichaelDeBoey could you run the tests?

Turns out the failure was resolved by doing the same steps to tar the directory on a mac instead of a windows... 🤷

**Comment by MichaelDeBoey** [comment]
@dvargas92495 Tests seem to fail on CI though.

**Comment by dvargas92495** [comment]
Whoops sorry! Left a typo on the rebase, could we rerun?

**Comment by dvargas92495** [comment]
Nice, @machour + @MichaelDeBoey PR is finally ready for review!

**Comment by dvargas92495** [comment]
Hey @mcansh, anything I could do on my end to get this ready for review?

**Comment by dvargas92495** [comment]
Hey @machour and @MichaelDeBoey I just rebased again could we get tests to run?

Once they run, is there anything I could do to help get it reviewed? Should I open a discussion? Trying to get a sense of what the blockers are

**Review by kentcdodds — APPROVED** [review]
Thank you for your hard work on this @dvargas92495. I'm a fan 👍 Thanks a bunch!

**Pull request merged**